fairscale/nn/wrap/auto_wrap.py

```diff
@@ -4,23 +4,72 @@
 # LICENSE file in the root directory of this source tree.
 
 import contextlib
-from typing import Any, Callable, Dict, Generator, List, Optional, Tuple
+from typing import Any, Callable, Dict, Generator, Optional, Set, Tuple, cast
 
 import torch.nn as nn
 
-from fairscale.nn.data_parallel.fully_sharded_data_parallel import FullyShardedDataParallel
 from fairscale.nn.misc import checkpoint_wrapper
 
-# Modules that don't wrap.
-FSDP_MODULE_EXCLUDE_WRAP = {nn.ModuleList, nn.ModuleDict}
-# Modules that we don't recurse down to their children.
-FSDP_MODULE_BLOCKLIST = {nn.MultiheadAttention}
+
+def should_wrap_default(
+    should_recurse: bool,
+    module: nn.Module,
+    unwrapped_params: int,
+    # These are customizable for this default policy function.
+    min_num_params: int = int(1e8),
+    blacklist: Optional[Set] = None,
+    exclude_wrap: Optional[Set] = None,
+) -> bool:
+    """Default policy function on with :func:`auto_wrap`.
+
+       Return if a module should be wrapped during :func:`auto_wrap`.
+
+       The first three parameters are used by :func:`auto_wrap`. If
+       you write a custom version of this policy function, your version
+       needs to at least accept the first three parameters and free
+       to do whatever you want in the function.
+
+    Args:
+       should_recurse (bool):
+           Indicate if this is called to make a decision on whether we
+           should recurse down a subgraph of the module structure.
+           If False, it means this function is called to make a decision
+           on whether we should wrap the said module.
+       module (nn.Module):
+           The module to be considered in this decision.
+       unwrapped_params (int):
+           The number of parameters yet to be wrapped in this module.
+
+       min_num_params (int):
+           Customizable policy input. It controls the size threshold
+           on how big should a module be to be considered wrapped.
+       blacklist (set):
```

Comment:
this needs to be renamed, there's a larger effort to remove blacklist/whitelist terminology across the company (more context, internal link only: https://fburl.com/z8pu53eq)

I also find the naming a bit unclear w.r.t. the `exclude_wrap` option.

How about we rename this to `force_leaf_modules`:
```python
force_leaf_modules (Set[Type[nn.Module]]): set of module types to
    keep as leaves, i.e., their children will never be wrapped.
```

Reply:
done. also changed exclude_wrap to exclude_wrap_modules to be similar to force_leaf_modules. Also, renamed the two constants accordingly.